Client/src/Components/SessionTimeout.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useContext, useState } from "react";
+
+import { Alert } from '@material-ui/lab';
+import { Snackbar } from '@material-ui/core';
+import { UserContext } from "../App";
+import { defaultUserAccountModel } from '../Models/Authentication/IUserAccountModel';
+
+import useInterval from 'react-useinterval';
+import { endUserSession } from "../Common/GenericHelpers";
+import { useHistory } from "react-router";
+import { loginRoute } from "../Common/Consts/Routes";
+
+export const SessionTimeOut = () => {
+
+    const { user, setUser } = useContext(UserContext);
+    const [seconds, setSeconds] = React.useState(user.sessionExpiration);
+    const [open, setOpen] = useState<boolean>(false)
+    const history = useHistory()
+
+    useInterval(() => {
+        if (seconds === 1) {
+            redirectToLogin()
+        }
+        else if (seconds < 30) {
+            setOpen(true)
+        }
+        setSeconds(seconds - 1)
+    }, 1000)
+
+    const redirectToLogin = async () => {
+        setUser(defaultUserAccountModel)
+        endUserSession()
```

Comment:
![image](https://user-images.githubusercontent.com/22376976/109552562-ef69a900-7a9f-11eb-84e7-380ed68539c7.png)
setUser already Removes the guy from storage, that makes the endUserSession() only make a call to signout, i think endUserSession can be removed and `await callSignOut()` can be directly called here

Reply:
Great